lib/tbot/config/service_kubernetes_v2.go

```diff
@@ -0,0 +1,161 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package config
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+	"gopkg.in/yaml.v3"
+
+	"github.com/gravitational/teleport/lib/tbot/bot"
+)
+
+var (
+	_ ServiceConfig = &KubernetesV2Output{}
+	_ Initable      = &KubernetesV2Output{}
+)
+
+const KubernetesV2OutputType = "kubernetes/v2"
+
+// KubernetesOutput produces credentials which can be used to connect to a
+// Kubernetes Cluster through teleport.
+type KubernetesV2Output struct {
+	// Destination is where the credentials should be written to.
+	Destination bot.Destination `yaml:"destination"`
+
+	// Roles is the list of roles to request for the generated credentials.
+	// If empty, it defaults to all the bot's roles.
+	Roles []string `yaml:"roles,omitempty"`
```

Comment:
I've been long thinking about deprecating the roles field - I'd be tempted to say that that should start now... it should be fairly to add back in if anybody complains, and perhaps this is a good way to gauge if anybody is actually seriously using it.

Reply:
Hmm, fine by me. Like you said, this is a good test case to see if anyone actually uses the feature.

Reply:
Alright, roles is now removed and we just pull in default roles every time.